# Imports

In [1]:
import pandas as pd
import numpy as np

---

# Installs - Carga de datos

In [2]:
installs = pd.read_csv("../../data/tp2/installs.csv")

/opt/anaconda/3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (4,15) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
installs.memory_usage(deep=True).sum() / 1024**2

228.32244110107422

In [4]:
exploracion = installs.dtypes.to_frame()

In [6]:
exploracion.columns = ["dtype"]

In [11]:
exploracion["tiene_nulos"] = installs.isna().any()

In [12]:
exploracion["todos_nulos"] = installs.isna().all()

In [13]:
exploracion["cantidad_nulos"] = len(installs) - installs.count()

In [14]:
exploracion["porcentaje_nulos"] = exploracion["cantidad_nulos"] / len(installs)

In [15]:
exploracion["porcentaje_nulos"] = exploracion["porcentaje_nulos"].map(lambda x: round(x*100, 2))

In [16]:
exploracion["maximo"] = installs.max()

In [17]:
exploracion["minimo"] = installs.min()

In [18]:
exploracion["valores_unicos"] = installs.nunique()

In [19]:
exploracion["porcentaje_valores_unicos"] = (exploracion["valores_unicos"] / len(installs)) *100 
exploracion["porcentaje_valores_unicos"] = exploracion["porcentaje_valores_unicos"].map(lambda x: round(x, 2))

In [20]:
exploracion

dtype  tiene_nulos  todos_nulos  cantidad_nulos  \
created              object        False        False               0   
application_id        int64        False        False               0   
ref_type              int64        False        False               0   
ref_hash              int64        False        False               0   
click_hash           object         True        False          480369   
attributed             bool        False        False               0   
implicit               bool        False        False               0   
device_countrycode    int64        False        False               0   
device_brand        float64         True        False          205068   
device_model        float64         True        False           26892   
session_user_agent   object         True        False           14839   
user_agent           object         True        False          150743   
event_uuid           object         True        False          378343   
kind                 object         True        False          378343   
wifi                 object         True        False          186682   
trans_id             object         True        False          472578   
ip_address            int64        False        False               0   
device_language     float64         True        False           27577   

                    porcentaje_nulos                   maximo  \
created                         0.00  2019-04-26 23:59:58.788   
application_id                  0.00                      360   
ref_type                        0.00      1891515180541284343   
ref_hash                        0.00      9223365873867066370   
click_hash                     99.76                      NaN   
attributed                      0.00                     True   
implicit                        0.00                     True   
device_countrycode              0.00      6287817205707153877   
device_brand                   42.59              9.22159e+18   
device_model                    5.58              9.22336e+18   
session_user_agent              3.08                      NaN   
user_agent                     31.31                      NaN   
event_uuid                     78.57                      NaN   
kind                           78.57                      NaN   
wifi                           38.77                     True   
trans_id                       98.14                      NaN   
ip_address                      0.00      9223362640454048358   
device_language                 5.73              9.21023e+18   

                                     minimo  valores_unicos  \
created             2019-04-18 00:00:01.560          480962   
application_id                            1             311   
ref_type                1494519392962156891               2   
ref_hash                     40621409780134          393565   
click_hash                              NaN            1142   
attributed                            False               2   
implicit                              False               2   
device_countrycode      6287817205707153877               1   
device_brand                     1.8923e+15            1084   
device_model                    7.96735e+14            6472   
session_user_agent                      NaN            4537   
user_agent                              NaN           13727   
event_uuid                              NaN          103083   
kind                                    NaN             183   
wifi                                  False               2   
trans_id                                NaN            4053   
ip_address                   33114277486038          281130   
device_language                 6.64561e+16             223   

                    porcentaje_valores_unicos  
created                                 99.89  
application_id                           0.06  
ref_type                                 0.00  
ref_hash

# Conclusiones
## created
Cargar como fecha, no hay nulos ni anómalos
## application_id
No hay nulos, cargar como int16.
## ref_type
No hay nulos, convertir a una única columna que sea "es_apple" o algo parecido.
## ref_hash
Cargarlo tal cual está, no hay nulos.
## click_hash
Casi todos los valores son nulos. No pareciera aportar mucho pero habría que probarla.
## attributed
Dejarla como está. No tiene nulos.
## implicit
Dejarla como está. No tiene nulos.
## device_countrycode
No cargarla, no hay nulos y se sabe que es Uruguay.
## device_brand
42% de nulos, habría que ver si aporta y cómo puede combinarse con otros datos.
## device_model
6% de nulos. Cargarla tal cual está.
## session_user_agent
Cargarla tal cual está, habría que ver cómo se combina con otros datos.
## user_agent
Cargarla como está, analizar si aporta y cómo combinarla.
## event_uuid
Tiene un 78% de nulos pero los valores se repiten. Habría que buscar si se puede relacionar con otros datos.
## kind
Tiene un 78% de nulos (llamativamente la misma cantidad que __event_uuid__. Habría que analizar si hay algúna relación.
## wifi
39% de nulos, cargarla tal cual está.
## trans_id
Casi todos nulos, ver si los que no son nulos significan algo o aportan alguna información.
## ip_address
No tiene nulos por lo que puede usarse para relacionar con otros datos. Cargarlo tal cual está.
## device_language
Ver si aporta, tiene pocos nulos pero 223 es un número grande de idiomas posibles para dispositivos que se usan en Uruguay.